src/super_gradients/training/metrics/depth_estimation_metrics.py

```diff
@@ -0,0 +1,250 @@
+from typing import Tuple, Sequence, Union, Optional
+
+import torch
+from torch import Tensor
+from torchmetrics import MeanSquaredError, MeanSquaredLogError, MeanAbsoluteError, MeanAbsolutePercentageError, Metric
+from super_gradients.common.object_names import Metrics
+from super_gradients.common.registry import register_metric
+
+
+class DepthEstimationMetricMixin:
+    """
+    A mixin class providing common functionality for depth estimation metrics.
+
+    :param ignore_val: Value to be ignored when computing metrics.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        self.ignore_val = ignore_val
+        self.apply_sigmoid = apply_sigmoid
+
+    def process_preds_and_target(self, preds: Union[Tensor, Sequence[Tensor]], target: Tensor) -> Tuple[Tensor, Tensor]:
+        """
+        Process predictions and target tensors for depth estimation metrics:
+        - If a sequence is returned by the model -> sets preds to the first element
+        - Squeezes the dummy dimension (i.e dim1) when preds.ndim == 4
+        - Applies sigmoid to preds if apply_sigmoid is True
+        - Removes entries to ignore where ignore_val is present in target
+
+        :param preds: Model predictions, either a tensor or a sequence of tensors.
+        :param target: Ground truth depth map.
+        :return: Processed predictions and target tensors.
+        """
+
+        if isinstance(preds, Sequence):
+            preds = preds[0]
+
+        if preds.ndim == 4:
+            preds = preds.squeeze(1)
+
+        if self.apply_sigmoid:
+            preds = torch.sigmoid(preds)
+
+        if self.ignore_val is not None:
+            non_ignored = preds != self.ignore_val
+            preds = preds[non_ignored]
+            target = target[non_ignored]
+
+        return preds, target
+
+
+@register_metric(Metrics.MeanSquaredErrorWithIgnored)
+class MeanSquaredErrorWithIgnored(MeanSquaredError, DepthEstimationMetricMixin):
+    """MeanAbsoluteError, MeanAbsolutePercentageError
+    Mean Squared Error metric for depth estimation with support for ignored values.
+
+    :param squared: If True returns MSE value, if False returns RMSE value.
+    :param ignore_val: Value to be ignored when computing the metric.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, squared: bool = True, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        super().__init__(squared=squared)
```

Comment:
The first `super().__init__` refers to the first class (`MeanSquaredError` here) right ?
I think in case of double inheritence it would be better to be more explicit about every init: `MeanSquaredError.__init__`

Reply:
Note: this may not be relevant if we go for something proposed in the next comment (maybe check it first)

Reply:
sure, updated in b86a122ac667ca07f5fe149d0fe795b4ca4e727d